state/factory/workingsetstore.go

```diff
@@ -33,24 +36,49 @@ func (store *workingSetStoreCommon) Filter(ns string, cond db.Condition, start,
 }
 
 func (store *workingSetStoreCommon) WriteBatch(bat batch.KVStoreBatch) error {
-	return store.flusher.KVStoreWithBuffer().WriteBatch(bat)
+	if err := store.flusher.KVStoreWithBuffer().WriteBatch(bat); err != nil {
```

Comment:
It should could work well. This way maybe more clear:
```
if store.commited {
    return store.flusher.BaseKVStore().WriteBatch(bat)
}
return store.flusher.KVStoreWithBuffer().WriteBatch(bat)
```

Reply:
we may need an extra lock here